kong/api/routes/upstreams.lua

```diff
@@ -160,4 +103,25 @@ return {
     end
   },
 
+  ["/upstreams/:name_or_id/targets/:target"] = {
+    before = function(self, dao_factory, helpers)
+      crud.find_upstream_by_name_or_id(self, dao_factory, helpers)
+      self.params.upstream_id = self.upstream.id
+    end,
+
+    DELETE = function(self, dao_factory)
+      balancer.clean_history(self.params.upstream_id, dao_factory)
+
+      -- this is just a wrapper around POSTing a new target with weight=0
+      local target  = self.params
+      target.weight = 0
+
+      local data, err = dao_factory.targets:insert(target)
+      if err then
+        return app_helpers.yield_error(err)
+      end
+
+      return responses.send_HTTP_NO_CONTENT()
```

Comment:
Wondering if we should not send `HTTP 201` still here. Yes, this is a DELETE and stays consistent with the rest of the API, but it isn't a traditional DELETE endpoint. It is known that this is a sugar method. Might bring confusion, but might also raise awareness about targets. However, this is mostly internal implementation details, so probably good to hide it away.